revprxy.yaml

```diff
@@ -0,0 +1,160 @@
+- hosts: 34.121.46.70
+  gather_facts: no
+  remote_user: ubuntu
+  become: true
+
+
+  tasks:
+    - name: ensure nginx is at the latest version
+      apt:
+        name=nginx
+        state=latest
+        update_cache=yes
+
+    - name: start nginx
+      service:
+        name: nginx
+        state: started
+
+    - name: create CA key
+      openssl_privatekey:
+        path: /root/CA_key.pem
+      register: ca_key
+
+    - name: create the CA CSR
+      openssl_csr:
+        path: /root/CA.csr
+        privatekey_path: "{{ ca_key.filename }}"
+        common_name: 34.121.46.70
+      register: ca_csr
+
+    - name: sign the CA CSR
+      openssl_certificate:
+        path: /root/CA.crt
+        csr_path: "{{ ca_csr.filename }}"
+        privatekey_path: "{{ ca_key.filename }}"
+        provider: selfsigned
+      register: ca_crt
+
+    - name: create host CSR signing key
+      openssl_privatekey:
+        path: /root/example_com_host_key.pem
+      register: example_com_key
+
+    - name: create the CSR for Nginx host
+      openssl_csr:
+        path: /root/example_com.csr
+        privatekey_path: "{{ example_com_key.filename }}"
+        common_name: 34.121.46.70
+        subject_alt_name: 'IP:34.121.46.70,DNS:www.example.com'
+      register: example_com_csr
+
+
+    - name: sign the CSR for the Nginx host
+      openssl_certificate:
+        path: /root/example_com.crt
+        csr_path: "{{ example_com_csr.filename }}"
+        provider: ownca
+        ownca_path: "{{ ca_crt.filename }}"
+        ownca_privatekey_path: "{{ ca_key.filename }}"
+      register: example_com_crt
+
+
+    - name: Creating a Configuration Snippet Pointing to the SSL Key and Certificate
+      blockinfile:
+        path: "/etc/nginx/snippets/self-signed.conf"
+        create: yes
+        block: |
+          ssl_certificate /root/example_com.crt;
+          ssl_certificate_key /root/example_com_host_key.pem;
+
+
+    - name: Creating a Configuration Snippet with Strong Encryption Settings
+      blockinfile:
+        path: "/etc/nginx/snippets/ssl-params.conf"
+        create: yes
+        block: |
+          ssl_protocols TLSv1.3;
+          ssl_prefer_server_ciphers on;
```

Comment:
keep this as external file and use template or copy module for this

Reply:
Done.